programs/mango-v3-reimbursement/src/instructions/reimburse.rs

```diff
@@ -0,0 +1,129 @@
+use anchor_lang::{__private::bytemuck, prelude::*};
+use anchor_spl::token::{self, Mint, Token, TokenAccount};
+
+use crate::state::{Group, ReimbursementAccount, Table};
+
+#[derive(Accounts)]
+#[instruction(token_index: usize)]
+pub struct Reimburse<'info> {
+    #[account (
+        constraint = group.load()?.has_reimbursement_started()
+    )]
+    pub group: AccountLoader<'info, Group>,
+
+    #[account(
+        mut,
+        constraint = group.load()?.vaults[token_index] == vault.key()
```

Comment:
btw, `address = group.load()?.vaults[token_index]` would do the same and I think new anchor can use those constraints for auto-derivation